src/sigma/coin.h

```diff
@@ -19,7 +19,7 @@ enum class CoinDenomination : std::uint8_t {
     SIGMA_DENOM_10 = 3,
     SIGMA_DENOM_25 = 6,
     SIGMA_DENOM_100 = 4,
-    SIGMA_ERROR = 7
+    SIGMA_ERROR = 7 
```

Comment:
This is not the right way to fix. The reason is we introduce a new possible value for denomination. The problems is not adding a new value but adding a new value that is not valid while any other values that not valid already invalid value.

Could you explain what you the problems you are trying to solve?

Reply:
Trying to introduce a default value for denomination, so when a `HDMint` object is initialized or set to NULL we reset the denomination value.

Reply:
Take a look on `boost::optional` (https://www.boost.org/doc/libs/1_65_1/libs/optional/doc/html/index.html). It should solve your problems without introduce a new special value for denomination.